crates/turborepo-lib/src/turbo_json/parser.rs

```diff
@@ -0,0 +1,556 @@
+}
+
+trait WithText {
+    fn add_text(&mut self, text: Arc<str>);
+}
+
+impl<T> WithText for Spanned<T> {
+    fn add_text(&mut self, text: Arc<str>) {
+        self.text = Some(text);
+    }
+}
+
+impl<T: WithText> WithText for Option<T> {
+    fn add_text(&mut self, text: Arc<str>) {
+        if let Some(inner) = self {
+            inner.add_text(text);
+        }
+    }
+}
+
+impl<T: WithText> WithText for Vec<T> {
+    fn add_text(&mut self, text: Arc<str>) {
+        for item in self {
+            item.add_text(text.clone());
+        }
+    }
+}
+
+impl WithText for RawTurboJson {
+    fn add_text(&mut self, text: Arc<str>) {
+        self.extends.add_text(text.clone());
+        self.global_dependencies.add_text(text.clone());
+        self.global_env.add_text(text.clone());
+        self.global_pass_through_env.add_text(text.clone());
+        self.pipeline.add_text(text.clone());
+    }
+}
+
+impl WithText for Pipeline {
+    fn add_text(&mut self, text: Arc<str>) {
+        for (_, task) in self.0.iter_mut() {
+            task.add_text(text.clone());
+        }
+    }
+}
+
+impl WithText for RawTaskDefinition {
+    fn add_text(&mut self, text: Arc<str>) {
+        self.depends_on.add_text(text.clone());
+        self.dot_env.add_text(text.clone());
+        self.env.add_text(text.clone());
+        self.inputs.add_text(text.clone());
+        self.pass_through_env.add_text(text.clone());
+        self.persistent.add_text(text.clone());
+        self.outputs.add_text(text.clone());
+        self.output_mode.add_text(text.clone());
+    }
+}
+
+impl RawTurboJson {
+    // A simple helper for tests
+    #[cfg(test)]
+    pub fn parse_from_serde(value: serde_json::Value) -> Result<RawTurboJson, Error> {
+        let json_string = serde_json::to_string(&value).expect("should be able to serialize");
+        Self::parse(&json_string, "turbo.json")
+    }
+    /// Parses a turbo.json file into the raw representation with span info
+    /// attached.
+    ///
+    /// # Arguments
+    ///
+    /// * `text`: The text contents of the turbo.json file
+    /// * `file_path`: The path to the turbo.json file. Just used for error
+    ///   display, so doesn't need to actually be a correct path.
+    ///
+    /// returns: Result<RawTurboJson, Error>
+    pub fn parse(text: &str, file_path: &str) -> Result<RawTurboJson, Error> {
+        let result = deserialize_from_json_str::<RawTurboJson>(
+            text,
+            JsonParserOptions::default().with_allow_comments(),
+        );
+
+        if !result.diagnostics().is_empty() {
+            let diagnostics = result
+                .into_diagnostics()
+                .into_iter()
+                .map(|d| d.with_file_source_code(text).with_file_path(file_path))
+                .collect();
+
+            return Err(Error::Parse { diagnostics });
+        }
+
+        let mut turbo_json = result
+            .into_deserialized()
+            .expect("should have turbo.json if no errors");
```

Comment:
So it's unclear. I'm not sure what to do here actually because we could error, but then the user will get a generic "there's a syntax error in your turbo.json" but without any actual indication of what that error is (since there's no diagnostics printed). It might be better to panic here and let the user report that as a bug so we can get an idea of when this may occur.